src/MyLinkedList.java

```diff
@@ -0,0 +1,123 @@
+import org.w3c.dom.Node;
+
+import java.util.LinkedList;
+
+public class MyLinkedList<E> implements List<E> {
+
+    private Node<E> firstNode;
+    private Node<E> lastNode;
+    private int size = 0;
+
+
+    @Override
+    public void add(E value) {
+        Node<E> node = new Node(value, null, null);
+        if (isEmpty()) {
+            firstNode = node;
+            lastNode = node;
+        } else {
+            node.prev = lastNode;
+            lastNode.next = node;
+            lastNode = node;
+        }
+        size++;
+    }
+
+    @Override
+    public void add(E value, int index) {
+        Node<E> node = new Node(value, null, null);
+        if (isEmpty()) {
+            firstNode = node;
+            lastNode = node;
+        }
+        else{
+
+            node.prev=findIndex(index-1);
+            findIndex(index-1).next=node;
+            node.next=findIndex(index);
+            findIndex(index).prev = node;
+
+        }
+        size++;
+    }
+
+    @Override
+    public void addAll(List<E> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+    }
+
+
+    public Node<E> findIndex(int index) {
+        checkSize(index);
+        Node<E> node = firstNode;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return node;
+    }
+
+    public E get(int index) {
+        return findIndex(index).item;
+
+    }
+
+    @Override
+    public void set(E value, int index) {
+        findIndex(index).item = value;
+    }
+
+    @Override
+    public E remove(int index) {
```

Comment:
try doesn't use method findIndex() all time; 
this worsens the performance of work

Reply:
we need to rewrite this method without often calling findIndex(index) . If you don't know how to do it, write me in slack